net/http/authn/authn.go

```diff
@@ -0,0 +1,100 @@
+package authn
+
+import (
+	"context"
+	"encoding/hex"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"chain/core/accesstoken"
+	"chain/errors"
+	"chain/net/authz"
+)
+
+const tokenExpiry = time.Minute * 5
+
+type API struct {
+	Tokens           *accesstoken.CredentialStore
```

Comment:
Putting this concrete type here takes a hard dependency from chain/net/http/authn on chain/core/accesstoken. We shouldn't do that — as a "general-purpose" library, packages outside of chain/core/... shouldn't depend on core packages.

Prob the right thing to do is define a one-method interface here that can be satisfied by *accesstoken.CredentialStore.

Maybe also, as a potential TODO item, is to take all the code in that package (chain/core/accesstoken) and move it all in here.